src/query/list_file.c

```diff
@@ -1918,8 +1917,7 @@ qfile_fast_val_tuple_to_list (THREAD_ENTRY * thread_p, QFILE_LIST_ID * list_id_p
     }
 
   /* list_id maintainance stuff */
-  offset = ((tuple_length + list_id_p->last_offset) > DB_PAGESIZE ? DB_PAGESIZE : tuple_length);
-  qfile_add_tuple_to_list_id (list_id_p, page_p, tuple_length, offset);
+  qfile_add_tuple_to_list_id (list_id_p, page_p, tuple_length, MIN (tuple_length, qfile_Max_tuple_page_size));
```

Comment:
add assert (page_p - cur_page_p <= DB_PAGESIZE);
tuple_length is already checked and is less than qfile_Max_tuple_page_size (which is same value as QFILE_MAX_TUPLE_SIZE_IN_PAGE).

Same for qfile_fast_intval_tuple_to_list.